packages/frontend/src/components/RegisterForms/VerifyEmailForm.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+    Anchor,
+    Button,
+    PinInput,
+    Stack,
+    Text,
+    Title,
+    UnstyledButton,
+} from '@mantine/core';
+import { isNotEmpty, useForm } from '@mantine/form';
+import React, { FC, useEffect, useState } from 'react';
+import Countdown, { zeroPad } from 'react-countdown';
+import {
+    useEmailStatus,
+    useOneTimePassword,
+    useVerifyEmail,
+} from '../../hooks/useEmailVerification';
+import { useApp } from '../../providers/AppProvider';
+import LoadingState from '../common/LoadingState';
+
+const VerifyEmailForm: FC<{ isLoading?: boolean }> = ({ isLoading }) => {
+    const { health, user } = useApp();
+    const { mutate: verifyCode, isLoading: verificationLoading } =
+        useVerifyEmail();
+    const { data, isLoading: statusLoading } = useEmailStatus();
+    const { mutate: sendVerificationEmail, isLoading: emailLoading } =
+        useOneTimePassword();
+    const form = useForm<{ code: string }>({
+        initialValues: {
+            code: '',
+        },
+        validate: {
+            code: isNotEmpty('This field is required.'),
+        },
+    });
+    const errorMessage = form.errors.code;
+    const expirationTime = data?.otp?.expiresAt || new Date();
+    const loadingState =
+        statusLoading || emailLoading || health.isLoading || isLoading;
+
+    useEffect(() => {
+        if (data?.otp && data?.otp.numberOfAttempts > 0) {
+            const remainingAttempts = 5 - data.otp.numberOfAttempts;
+            const message = data.otp.isExpired
+                ? 'Your one-time password expired. Please resend a verification email.'
+                : data.otp.numberOfAttempts < 5
+                ? `The code doesn't match the one we sent you. You have ${remainingAttempts} attempt${
+                      remainingAttempts > 1 ? 's' : ''
+                  } left.`
+                : "Hmm that code doesn't match the one we sent you. You've already had 5 attempts, please resend a verification email and try again.";
+            form.setFieldError('code', message);
+        } else {
+            form.clearFieldError('code');
+        }
+    }, [data, form.setFieldError, form.clearFieldError]);
+
+    if (loadingState) {
+        return <LoadingState title="" />;
+    }
+
+    return (
+        // FIXME: use Mantine sizes for width
+        <Stack spacing="md" justify="center" align="center" w={300} mx="auto">
```

Comment:
with the current scale I think this should be 320
```
        one: rem(1),
        two: rem(2),
        xxs: rem(4),
        xs: rem(8),
        sm: rem(12),
        md: rem(16),
        lg: rem(20),
        xl: rem(24),
        xxl: rem(32),
        '3xl': rem(40),
        '4xl': rem(48),
        '5xl': rem(64),
        '6xl': rem(80),
        '7xl': rem(96),
        '8xl': rem(128),
        '9xl': rem(160),
```

we might need to make a scalable utility for size scales (in extreme cases like this)

Reply:
leaving hardcoded widths for now, added comments to find them easily!